fix:Save card on lost focus if not empty

codingseb opened the pull request:
# Corrections
Correct #1149, #1123, #1091 and #1100 behavior.

New behaviors when click outside a new card:
* if content is empty cancel the card
* if content is not empty validate the card to not lose it

New behavior when click outside a card in edition :
* Validate the edition to not lose the work done in the card if the note change.

This should avoid a lot of frustration 

# Tests 
I tested it on my vault and it works well.

codingseb committed:
feat: Implement intelligent click-outside save/cancel for ItemContent

codingseb committed:
refactor: Refine click-outside behavior for new item creation in ItemForm

codingseb committed:
Translate comments in english

lucaskuzma commented:
Please merge! 😬🙏 @mgmeyers

rscircus commented:
Hi, I run into this daily. How can I help to get this merged?

rscircus commented:
@codingseb - I'll test it on my vault and add the results in this comment today. That's the least thing I can do for helping on the robustness side. Let me know if you need a second pair of eyes for the rest of the process here. :)

codingseb committed:
feat: Saves edits on view/tab close

Implements a global save mechanism to persist edits
before a Kanban view is closed or a tab is changed.

This ensures no data loss by triggering a 'save-all-editing-items'
event, which is listened for by both ItemContent and ItemForm
components to save any pending changes.

codingseb committed:
fix: Prevent item content/form closure when clicking menu items

Previously, clicking on menu elements (e.g., dropdowns, context menus) associated with an item could erroneously trigger the `useOnclickOutside` handler. This could lead to premature saving or cancellation of an item's editing state.

By adding `menu` and `menu-item` to the `ignoreClass` list for the `useOnclickOutside` hook, these interactions are now correctly disregarded, ensuring a smoother user experience.
